logstreams/src/main/java/io/zeebe/logstreams/impl/service/LogStreamService.java

```diff
@@ -62,28 +61,44 @@
   private Dispatcher writeBuffer;
   private LogStorageAppender appender;
 
-  public LogStreamService(final LogStreamBuilder builder) {
-    logName = builder.getLogName();
-    partitionId = builder.getPartitionId();
-    serviceContainer = builder.getServiceContainer();
-    onCommitPositionUpdatedConditions = builder.getOnCommitPositionUpdatedConditions();
-    commitPosition = builder.getCommitPosition();
-    maxFrameLength = ByteValue.ofBytes(builder.getMaxFragmentSize());
+  public LogStreamService(
+      final ServiceContainer serviceContainer,
+      final ActorConditions onCommitPositionUpdatedConditions,
+      final String logName,
+      final int partitionId,
+      final ByteValue maxFrameLength,
+      final Position commitPosition,
+      final LogStorage logStorage) {
+    this.serviceContainer = serviceContainer;
+    this.onCommitPositionUpdatedConditions = onCommitPositionUpdatedConditions;
+    this.logName = logName;
+    this.partitionId = partitionId;
+    this.maxFrameLength = maxFrameLength;
+    this.commitPosition = commitPosition;
+    this.logStorage = logStorage;
   }
 
   @Override
   public void start(final ServiceStartContext startContext) {
-    commitPosition.setVolatile(INVALID_ADDRESS);
+    if (logStorage == null) {
+      logStorage = logStorageInjector.getValue();
+    }
 
+    try {
+      logStorage.open();
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+
+    commitPosition.setVolatile(INVALID_ADDRESS);
     serviceContext = startContext;
-    logStorage = logStorageInjector.getValue();
     this.reader = new BufferedLogStreamReader(this);
     setCommitPosition(reader.seekToEnd());
```

Comment:
This would be still not really performant right. Would it be possible to set the commit position on creation? We get probably the position from atomix right?